### PR TITLE
xray: aws xray tracer span data bug fix

### PR DESCRIPTION
**Pull request by suniltheta**
Commit Message: fix AWS X-Ray tracer bug where span error was not properly reported.
Additional Description: Remove 'error' tag in the trace span from getting reported inside annotations section of xray tracer segment data and instead move it inside "Errors and faults" section as per the [xray documentation](https://docs.aws.amazon.com/xray/latest/devguide/xray-api-segmentdocuments.html#api-segmentdocuments-errors). Bug description available at https://github.com/aws/aws-app-mesh-roadmap/issues/344 .
Risk Level: Low
Testing: Unit tests added
Docs Changes: No
Release Notes: Bug info
Platform Specific Features: No

### CODE REVIEW

**Comment by suniltheta** [comment]
Hi @tonya11en this PR is ready for review. Thank you for your time.

**Comment by tonya11en** [comment]
Sorry, I'm on vacation until 7/12. I can take a look as soon as I get back.

**Comment by suniltheta** [comment]
Sorry about that, no hurry. This can wait.

File: source/extensions/tracers/xray/daemon.proto [hunk 1]
```diff
@@ -17,6 +17,9 @@ message Segment {
   string origin = 9;
   string parent_id = 6;
   google.protobuf.Struct aws = 10;
+  bool error = 11;
```
Comment (tonya11en) [inline source/extensions/tracers/xray/daemon.proto:20]: Can you add some 1-liner comments explaining what each message is. I know there's a link to the documentation there, but it's still seems appropriate to have some information here.

File: test/extensions/tracers/xray/tracer_test.cc [hunk 2]
```diff
@@ -90,13 +95,236 @@ TEST_F(XRayTracerTest, SerializeSpanTest) {
...
+    ASSERT_STREQ(expected_http_url, s.http().request().fields().at("url").string_value().c_str());
+    ASSERT_STREQ(
+        expected_user_agent,
+        s.http().request().fields().at(Tracing::Tags::get().UserAgent).string_value().c_str());
+    ASSERT_FALSE(s.fault());   /*server error*/
+    ASSERT_TRUE(s.error());    /*client error*/
+    ASSERT_TRUE(s.throttle()); /*request throttled*/
+    ASSERT_DOUBLE_EQ(expected_status_code,
+                     s.http().response().fields().at(Tracing::Tags::get().Status).number_value());
+  };
```
Comment (tonya11en) [inline test/extensions/tracers/xray/tracer_test.cc:232]: Is it possible to dedup this across some of the tests? Seems like you can capture a struct with the expected values and pass it into a func with all the common assertions.

Also, why ASSERT vs. EXPECT?

File: test/extensions/tracers/xray/tracer_test.cc [hunk 3]
```diff
@@ -90,13 +95,236 @@ TEST_F(XRayTracerTest, SerializeSpanTest) {
...
+  constexpr auto expected_origin_name = "AWS::Service::Proxy";
+  constexpr auto expected_aws_key_value = "test_value";
+  constexpr auto expected_operation_name = "Create";
+  constexpr auto expected_http_method = "POST";
+  constexpr auto expected_http_url = "/first/second";
+  constexpr auto expected_user_agent = "Mozilla/5.0 (Macintosh; Intel Mac OS X)";
+
+  auto on_send = [&](const std::string& json) {
+    ASSERT_FALSE(json.empty());
+    daemon::Segment s;
```
Comment (tonya11en) [inline test/extensions/tracers/xray/tracer_test.cc:258]: Does it make sense to also validate this proto?

File: source/extensions/tracers/xray/daemon.proto (hunk 1 above)
Comment (suniltheta) [reply source/extensions/tracers/xray/daemon.proto:33]: Done.

File: test/extensions/tracers/xray/tracer_test.cc (hunk 2 above)
Comment (suniltheta) [reply test/extensions/tracers/xray/tracer_test.cc]: There were 6 checked that was common across all these functions. As you suggested I created a func and extracted out the repeated code.

Yes actually EXPECT makes more sense, and I have transformed most of the ASSERT into EXPECT.

File: test/extensions/tracers/xray/tracer_test.cc (hunk 3 above)
Comment (suniltheta) [reply test/extensions/tracers/xray/tracer_test.cc:228]: Yes true, I added validate on this proto in the new revision.  Thanks for this tip.

**Review by tonya11en — APPROVED** [review]
thanks and sorry for the delay

**Comment by suniltheta** [comment]
Hi @lizan, can you help merge this PR?

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**